geode-core/src/main/java/org/apache/geode/internal/cache/execute/AbstractExecution.java

```diff
@@ -49,7 +49,7 @@
  * @since GemFire 5.8LA
  *
  */
-public abstract class AbstractExecution implements InternalExecution {
+public abstract class AbstractExecution<IN, OUT, AGG> implements InternalExecution<IN, OUT, AGG> {
```

Comment:
This is better. I get `IN` and `OUT`, but what's an `AGG`?

Reply:
@pivotal-jbarrett  `AGG` is aggregator as it is in the `Execution` interface. `OUT` and `AGG` correspond with the `T` and `S` on the `ResultCollector` interface.

Reply:
looks like `T` is the result of single execution, `S` is a collection of them. Probably we can keep them as it is now, maybe later, we can see if we can simplify this to be just `IN` and `OUT`, and `T` to be just a `Collection<S>`.